### DISCUSSION TITLE
[Question] Can you get the app "hash"?

### DISCUSSION THREAD

**Discussion by BeyondEvil** [nrwl/nx#12135] [General]
Without deep diving into Nx internals, I'm guessing Nx stores a hash of each app and its dependencies for the `affected` command to work.

Essentially this: https://nx.dev/concepts/how-caching-works

Is it possible to get that hash using either the CLI or Nx devkit/plugins?

Currently we have a pretty complex shell script (🤮 ) that ends in something like:

```sh
hash=$(git log -n 1 --format="%H" -- "${affected_files[@]}")
```

where `affected_files` contains the directories of an app and its dependencies (packages/libs).

The hash is used to tag the container that subsequently is built for that app.

**Comment by fpereira1** [comment]
I think what you need is this:  `nx print-affected --target=build | jq ".tasks | map({id: .id, hash: .hash})"`

Haven't been able to put this to work yet but I think it's correct
